stubs/JACK-Client/jack/__init__.pyi

```diff
@@ -196,12 +223,12 @@ class OwnPort(Port):
     def disconnect(self, other: str | Port | None = ...) -> None: ...
     def unregister(self) -> None: ...
     def get_buffer(self) -> _CBufferType: ...
-    def get_array(self) -> _NDArray: ...
+    def get_array(self) -> NDArray[numpy.float32]: ...
 
 class OwnMidiPort(MidiPort, OwnPort):
-    def __init__(self, *args: Any, **kwargs: Any) -> None: ...
-    def get_buffer(self) -> _CBufferType: ...
-    def get_array(self) -> _NDArray: ...
+    def __init__(self, port_ptr: _CDataBase, client: Client) -> None: ...
+    def get_buffer(self) -> NoReturn: ...
```

Comment:
These raise NotImplementedError. Returning NoReturn isn't helpful for users, since that will just make code calling these unreachable. Maybe mark them with `@abstractmethod` instead.

Reply:
But then you get `Cannot instantiate abstract class "OwnMidiPort" with abstract attributes "get_array" and "get_buffer"` when `OwnMidiPort` is *not* abstract.

```
    def get_buffer(self):
        """Not available for MIDI ports."""
        raise NotImplementedError('get_buffer() not available on MIDI ports')

    def get_array(self):
        """Not available for MIDI ports."""
        raise NotImplementedError('get_array() not available on MIDI ports')
```

it sounds more like "These don't exist on MIDI ports" than "Not yet implemented / subclass it and implement it". Basically 
```py
get_buffer = None
get_buffer = None
```
would be equivalent. `NotImplementedError` may not have been the best choice of error to raise from the library.